adc/read_vsys/power_status.c

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (c) 2023 Raspberry Pi (Trading) Ltd.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include "stdbool.h"
+#include "hardware/adc.h"
+#include "power_status.h"
+
+#if CYW43_USES_VSYS_GPIO_PIN
+#include "pico/cyw43_arch.h"
+#endif
+
+// Get power status
+// Must have called adc_init
+// On Pico W must also have called cyw43_arch_init
+void power_status(bool *mains_powered, float *voltage_result) {
```

Comment:
To me "mains" implies 240V, which we're **definitely** not powering Pico directly from! :joy: 
So perhaps `usb_powered` is a better name than `mains_powered`?

Reply:
Yes, I didn't really like it. I don't like "usb powered" either as you don't have to be powered by usb. It would probably be better to change it to "battery_powered".

Reply:
> as you don't have to be powered by usb

Although it probably is 90% of the time? :shrug:

Reply:
if you're using a power supply you'll be using usb 0% of the time